src/generate-types.ts

```diff
@@ -1270,6 +1272,29 @@ export function run(options: Options) {
     }
 
     if (isScalar(schema)) {
+      let isA: ts.ArrowFunction | undefined = undefined;
+      if (schema.enum) {
```

Comment:
sweet

I wonder if we could do the same things for all scalars using the maker and asking for isSuccess?

it would be a bit of a lie though as 'deadbeef' may or may not be some identifier. However this is the case for enums also. Anyhow we could use the maker for enum also if we are more confident in our little lie with them.

Reply:
Changed it to all scalars:
```yml
    laughs:
      type: string
      pattern: (h[aeiou])+
```

should result in:
```ts
Object.assign(typeLaughs, {
    name: "Laughs",
    definition: {
        type: "string",
        pattern: "(h[aeiou])+"
    },
    maker: makeLaughs,
    isA: (value: any) => makeLaughs(value).isSuccess()
})
```

Reply:
Could we just use `isSuccess()` for everything? and ditch the `instanceof` stuff completely?
That way we'd all support `oneOf` types :D

Reply:
> it would be a bit of a lie though as 'deadbeef' may or may not be some identifier

What do you mean by identifier?

Reply:
say you have hexadecimal ids like UserIds and AccountIds. Whether 'deadbeef' is an account id or a user id depends on how it originated and that we cannot figure out as strings do not have identity

Reply:
not to say that having unscoped identifiers is that brilliant idea to start with.